### ISSUE TITLE
Scale binding flips inverted domains

### ISSUE THREAD

**Issue by jakevdp** [vega/vega-lite#5386]
If you bind a selection to scales on an inverted domain, the domain flips on first interaction. Here is an example ([vega editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAmCGAutIC4zEgVwE4BsVTkQHoBjWLAZwDoArCgewDtIBfAGnCgFtyBrfSAAd6AS0bxIHCJACmjEvWhiA5vlARpADzWcNUeAE9BMgQEcMscSMTwRANxNS9UAGYiZOaAIAS9SjOEAdxksSV0NSAoyHBNUdGh6HjF8AG0AJgBWAAY2MCyAXRZw9nDIAzVIQ2MzCysbe0dXd08BAFkRGIoAfWMsLoBxWBwcJlZOEukKDxkSW1G40sFLCqrYqDF4ELshySgAIzEvVEjomQoxiCKWIA)):
```json
{
  "data": {"url": "data/cars.json"},
  "mark": "point",
  "encoding": {
    "x": {
      "type": "quantitative",
      "field": "Horsepower",
      "scale": {"domain": [250, 0]}
    },
    "y": {"type": "quantitative", "field": "Miles_per_Gallon"}
  },
  "selection": {
    "pan": {"type": "interval", "bind": "scales"}
  }
}
```
First reported in https://github.com/altair-viz/altair/issues/1693

**Comment by arvind** [comment]
Closing in favor of #5413 which has a few more specs that exhibit similar/related issues.

**Comment by arvind** [comment]
Actually, this issue occurs due to a different cause than the specs in #5413. In particular, selection-driven scale domains are read off the top-level/resolved selection signal which automatically sorts values in ascending order as part of the unioning/intersection strategy.

**Comment by arvind** [comment]
Fixed with #7163.
